Utils/src/L1NonPrefiringProbProducer.cc

```diff
@@ -0,0 +1,348 @@
+// system include files
+#include <memory>
+
+#include "FWCore/Framework/interface/Frameworkfwd.h"
+#include "FWCore/Framework/interface/global/EDProducer.h"
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/Utilities/interface/StreamID.h"
+
+#include "DataFormats/Math/interface/deltaR.h"
+#include "DataFormats/PatCandidates/interface/Jet.h"
+#include "DataFormats/PatCandidates/interface/Photon.h"
+#include "DataFormats/PatCandidates/interface/Muon.h"
+#include "TFile.h"
+#include "TH2.h"
+#include "TF1.h"
+
+#include "TreeMaker/Utils/interface/EnergyFractionCalculator.h"
+
+#include <iostream>
+#include <string>
+#include <vector>
+#include <array>
+#include <cmath>
+
+//based on: https://github.com/lathomas/cmssw/blob/L1Prefiring_9_4_9/L1Prefiring/EventWeightProducer/plugins/L1NonPrefiringProbProducer.cc
```

Comment:
update this comment to point to the latest official version w/ muon prefiring

Reply:
Done, will appear in next commit